distributed/scheduler.py

```diff
@@ -1354,6 +1362,7 @@ def __init__(self, key: str, run_spec: object, state: TaskStateState):
         self.metadata = {}
         self.annotations = {}
         self.erred_on = set()
+        self._attempt = 0
```

Comment:
IIUC, this means that if we re-create a task, e.g. by re-running a workload after it has been forgotten, its attempt count would be reset to 0. This would mean that the attempt counter is _not_ unique. We should use a global counter instead that we also actively increment. `transition_counter` would be one possible candidate if we make sure to increment it every time we update `_attempt`.

Reply:
sure but I'd use a different name for this then since it's no longer an attempt